### DISCUSSION TITLE
Question about Flashing Firmware

### DISCUSSION THREAD

**Discussion by M7Fa** [CapibaraZero/fw#48] [Q&A]
Hello,

First of all, this is an awesome project and I'm really looking forward to building this.  I'm working with a breadboard right and have done a few things with Arduino Uno using sketches; so this is my first time flashing the firmware.  I bought the Arduino Nano 33 IoT (based on the link from the Arduino Article on how to setup the project -- https://projecthub.arduino.cc/andreockx/capibarazero-a-multifunctional-security-and-hacking-tool-15a318).  However, I'm completely unable to get esptool to flash the firmware.  Using the Arduino IDE I can upload a sketch (like blink).  I have the board in bootloader mode and it's assigned a COM port (Windows).  Do, I need a specialized tool to flash the firmware through the SWD pins or do I just have the wrong board?

Thanks,
M7

**Comment by andreock** [comment]
Hi! Arduino Nano 33 iot is not supporterd, I don't know why article go to Arduino Nano 33 iot link under the Arduino Nano ESP32 component. I will Fix later. The correct board is this one: https://store.arduino.cc/products/nano-esp32-with-headers?queryID=7281fc19441a125e4e3746f134f515f2

Inviato da Proton Mail Android

-------- Messaggio originale --------
01/11/24 18:05, M7Fa  ha scritto:
